### PR TITLE
test: Add tests for typography, layout, product and generic template

### PR DESCRIPTION
**Pull request by uarlouski**
## Changes

* Enable ESlint for playwright tests
* Added tests for typography, layout, product and generic template

### CODE REVIEW

File: playwright/pages/vital-cards.ts [hunk 1]
```diff
@@ -1,113 +1,127 @@
...
-        hasButton: true,
-    },
-    {
-        id: 'HeroSecondaryButtonRightContentTop',
-        hasButton: true,
-    },
-    {
-        id: 'HeroSecondaryButtonRightContentCentered',
-        hasButton: true,
+import { VitalElement, VitalPage } from "./vital-page";
```
Comment (dewen) [inline playwright/pages/vital-cards.ts:1]: Add open source license declaration header, see
https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/playwright/pages/accordion-page.ts

Comment (dewen) [reply playwright/pages/vital-cards.ts:1]: indent with 2 spaces instead of 4.

File: playwright/pages/vital-cards.ts [hunk 2]
```diff
@@ -1,113 +1,127 @@
...
+            {
+                id: 'HeroSecondaryButtonRightContentCentered',
+                hasButton: true,
+            }
+        ]
     }
-]
+
+    getElements(): VitalElement[] {
+        return this.vitalCards.map((card) => { return {id: card.id}});
```
Comment (dewen) [inline playwright/pages/vital-cards.ts:122]: `return this.vitalCards.map(card => ({id: card.id}));`

File: playwright/tests/smoke/vital-typography.spec.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  const typographyPage: VitalTypographyPage = new VitalTypographyPage()
+
+  typographyPage.typographyElements.filter((tElement) => tElement.isClickable).forEach((tElement) => {
+    test(`Should click on ${tElement.id} typography element`, async ({ page }) => {
+      await page.goto(typographyPage.relativeUrl)
+      await page.waitForLoadState()
+      const element = page.getByTestId(tElement.id)
+                          .locator('a')
+      await element.click({ noWaitAfter: false })
+      expect(page.url().endsWith('/test/')).toBeTruthy()
```
Comment (dewen) [inline playwright/tests/smoke/vital-typography.spec.ts:14]: @uarlouski  just curious why it's all end with test?

**Review by dewen — APPROVED** [review]

**Review by VancheeZze — APPROVED** [review]

**Pull request merged**